cli/azd/internal/errors.go

```diff
@@ -0,0 +1,49 @@
+package internal
+
+// ErrorWithSuggestion is a custom error type that includes a suggestion for the user
+type ErrorWithSuggestion struct {
+	Suggestion string
+	Err        error
+}
```

Comment:
What happen if the error that is bubbled up contains multiple ErrorWithSuggestion?  Do we print the list of all suggestions? 

The reason why I ask is because usually I expect some style in the form of:

error: error2: error3: errorN

Now, if multiple errors have a suggestion, how would it look?

Reply:
This isn't a use case that was handled previously.  If we want to support this type of behavior we likely should implement the inclusion of the "Suggestion" message at the time the `Error()` function is called so that it can unwrap all inner errors.

We can discuss this as a follow-up to this PR.

Reply:
We should keep it as-is for now. We did discuss briefly this possibility during the original implementation, but haven't seen an actual use case for it yet.

Reply:
I would recommend keeping this error on `azcli package` and let any other package to define its own error when needed.  Introducing an Internal error that surface packages use to return erros seems not like a good idea to me.
To me, Internal package is purely implementation details.

Reply:
@vhvb1989 The error here is for "azd app usage", not for a specific package like `azcli`. It needs to be inspected by the app (`cmd` level). I think `internal` fits this use-case currently, although we could come up with something similar to `pkg/azd` that `cmd/azd` uses if we think that is better.

I have been thinking about this for some time, but never quite found the right place for it. `internal` seems like a fine approach for me.

Reply:
Rolling back the constructor updates and just using raw structs for now.  Keep in `internal` namespace to be leveraged by other packages.

Reply:
@weikanglim , the error is currently defined in tools/azcli, within the azcli. It is used by other tools like dotnet , from the project and from cmd level:

![image](https://github.com/Azure/azure-dev/assets/24213737/6b77e10d-27f2-4274-8046-9704c5b5c5a6)

In my mind, internal is meant for cmd only.  But, I might be wrong :)